onnxruntime/contrib_ops/cpu/bert/attention.cc

```diff
@@ -238,8 +238,13 @@ Status Attention<T>::Compute(OpKernelContext* context) const {
       float* x = reinterpret_cast<T*>(scratch_data) + j * D;
       float* y = x;
 
-      for (int i = 0; i < D; i++)
-        y[i] = expf(x[i]);
+      float max = 0.f;
```

Comment:
Another way to get max (assume D > 0):
   max = x[0];
  for (i = 1; i < D; i++) {...
Otherwise, you might need add comments for case that all x[i] is negative.

Reply:
If all inputs are negative, e^x won't have instability issue, thus it is good to subtract 0 (max) from the value list.

Reply:
Please add some comments in code to avoid confusion.